### PR TITLE
Fix the computation of slice's output TV. 

### LINKED ISSUE #2552: RuntimeError: Tried to bind to a constant value: 7 as 1
Blocks https://github.com/Lightning-AI/lightning-thunder/issues/191

To reproduce, check out wjy/bind and run `python repro.py`. 

The backtrace is
```
FAILED thunder/tests/test_jit_general.py::test_litgpt_variants[cuda-falcon-7b-like] - RuntimeError: Tried to bind to a constant value: 7 as 1
Exception raised from bind_ at /Fuser/csrc/expr_evaluator.cpp:91 (most recent call first):
frame #0: nvfuser::nvfCheckFail(char const*, char const*, unsigned int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&) + 0xf3 (0x7f44a3409d9f in /usr/local/lib/python3.10/dist-packages/nvfuser-0.2.6+gitc29ae24-py3.10-linux-x86_64.egg/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
frame #1: nvfuser::ExpressionEvaluator::bind_(nvfuser::Val const*, dynamic_type::DynamicType<dynamic_type::Containers<std::vector>, nvfuser::StructHandle, nvfuser::Pointer, nvfuser::Opaque, at::Tensor, std::complex<double>, double, long, bool>, bool) + 0x1a9a (0x7f44a379a8fa in /usr/local/lib/python3.10/dist-packages/nvfuser-0.2.6+gitc29ae24-py3.10-linux-x86_64.egg/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
frame #2: nvfuser::ExpressionEvaluator::bind_(nvfuser::Val const*, dynamic_type::DynamicType<dynamic_type::Containers<std::vector>, nvfuser::StructHandle, nvfuser::Pointer, nvfuser::Opaque, at::Tensor, std::complex<double>, double, long, bool>, bool) + 0x1016 (0x7f44a3799e76 in /usr/local/lib/python3.10/dist-packages/nvfuser-0.2.6+gitc29ae24-py3.10-linux-x86_64.egg/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
frame #3: <unknown function> + 0x47c7cf (0x7f44a375c7cf in /usr/local/lib/python3.10/dist-packages/nvfuser-0.2.6+gitc29ae24-py3.10-linux-x86_64.egg/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
frame #4: nvfuser::FusionExecutor::runFusion(nvfuser::KernelArgumentHolder&, nvfuser::LaunchParams const&, nvfuser::CompileParams, std::vector<at::Tensor, std::allocator<at::Tensor> >) + 0x2da7 (0x7f44a376a087 in /usr/local/lib/python3.10/dist-packages/nvfuser-0.2.6+gitc29ae24-py3.10-linux-x86_64.egg/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
frame #5: <unknown function> + 0x67de91 (0x7f44a395de91 in /usr/local/lib/python3.10/dist-packages/nvfuser-0.2.6+gitc29ae24-py3.10-linux-x86_64.egg/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
frame #6: <unknown function> + 0x683721 (0x7f44a3963721 in /usr/local/lib/python3.10/dist-packages/nvfuser-0.2.6+gitc29ae24-py3.10-linux-x86_64.egg/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
frame #7: nvfuser::FusionKernelRuntime::runWithInputs(nvfuser::KernelArgumentHolder&) + 0xa9 (0x7f44a3963f79 in /usr/local/lib/python3.10/dist-packages/nvfuser-0.2.6+gitc29ae24-py3.10-linux-x86_64.egg/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
frame #8: nvfuser::FusionExecutorCache::runFusionWithInputs(c10::ArrayRef<c10::IValue> const&, std::optional<nvfuser::PrimDataType>, std::optional<signed char>) + 0x3fe (0x7f44a396f15e in /usr/local/lib/python3.10/dist-packages/nvfuser-0.2.6+gitc29ae24-py3.10-linux-x86_64.egg/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
frame #9: nvfuser::python_frontend::FusionDefinition::execute(c10::ArrayRef<c10::IValue> const&, std::optional<signed char>, bool, bool, bool) const + 0x38b (0x7f44a3b6183b in /usr/local/lib/python3.10/dist-packages/nvfuser-0.2.6+gitc29ae24-py3.10-linux-x86_64.egg/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
frame #10: <unknown function> + 0x1a4bce (0x7f44a3484bce in /usr/local/lib/python3.10/dist-packages/nvfuser-0.2.6+gitc29ae24-py3.10-linux-x86_64.egg/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
frame #11: <unknown function> + 0x21b8cf (0x7f44a34fb8cf in /usr/local/lib/python3.10/dist-packages/nvfuser-0.2.6+gitc29ae24-py3.10-linux-x86_64.egg/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
frame #12: <unknown function> + 0x2b1040 (0x7f44a3591040 in /usr/local/lib/python3.10/dist-packages/nvfuser-0.2.6+gitc29ae24-py3.10-linux-x86_64.egg/nvfuser/_C.cpython-310-x86_64-linux-gnu.so)
frame #13: <unknown function> + 0x15a10e (0x55e3305ae10e in /usr/bin/python3.10)
frame #14: _PyObject_MakeTpCall + 0x25b (0x55e3305a4a7b in /usr/bin/python3.10)
frame #15: <unknown function> + 0x168acb (0x55e3305bcacb in /usr/bin/python3.10)
frame #16: _PyEval_EvalFrameDefault + 0x198c (0x55e33059853c in /usr/bin/python3.10)
frame #17: <unknown function> + 0x16893e (0x55e3305bc93e in /usr/bin/python3.10)
frame #18: _PyEval_EvalFrameDefault + 0x2a27 (0x55e3305995d7 in /usr/bin/python3.10)
frame #19: _PyObject_FastCallDictTstate + 0xc4 (0x55e3305a3c14 in /usr/bin/python3.10)
frame #20: _PyObject_Call_Prepend + 0x5c (0x55e3305b986c in /usr/bin/python3.10)
frame #21: <unknown function> + 0x280700 (0x55e3306d4700 in /usr/bin/python3.10)
frame #22: _PyObject_MakeTpCall + 0x25b (0x55e3305a4a7b in /usr/bin/python3.10)
frame #23: _PyEval_EvalFrameDefault + 0x64e6 (0x55e33059d096 in /usr/bin/python3.10)
frame #24: _PyFunction_Vectorcall + 0x7c (0x55e3305ae9fc in /usr/bin/python3.10)
frame #25: _PyEval_EvalFrameDefault + 0x2a27 (0x55e3305995d7 in /usr/bin/python3.10)
frame #26: _PyFunction_Vectorcall + 0x7c (0x55e3305ae9fc in /usr/bin/python3.10)
frame #27: _PyEval_EvalFrameDefault + 0x2a27 (0x55e3305995d7 in /usr/bin/python3.10)
frame #28: _PyFunction_Vectorcall + 0x7c (0x55e3305ae9fc in /usr/bin/python3.10)
frame #29: _PyEval_EvalFrameDefault + 0x2a27 (0x55e3305995d7 in /usr/bin/python3.10)
frame #30: _PyFunction_Vectorcall + 0x7c (0x55e3305ae9fc in /usr/bin/python3.10)
frame #31: _PyEval_EvalFrameDefault + 0x2a27 (0x55e3305995d7 in /usr/bin/python3.10)
frame #32: _PyFunction_Vectorcall + 0x7c (0x55e3305ae9fc in /usr/bin/python3.10)
frame #33: _PyEval_EvalFrameDefault + 0x2a27 (0x55e3305995d7 in /usr/bin/python3.10)
frame #34: <unknown function> + 0x16893e (0x55e3305bc93e in /usr/bin/python3.10)
frame #35: _PyEval_EvalFrameDefault + 0x2a27 (0x55e3305995d7 in /usr/bin/python3.10)
frame #36: <unknown function> + 0x16893e (0x55e3305bc93e in /usr/bin/python3.10)
frame #37: _PyEval_EvalFrameDefault + 0x2a27 (0x55e3305995d7 in /usr/bin/python3.10)
frame #38: _PyObject_FastCallDictTstate + 0xc4 (0x55e3305a3c14 in /usr/bin/python3.10)
frame #39: _PyObject_Call_Prepend + 0x5c (0x55e3305b986c in /usr/bin/python3.10)
frame #40: <unknown function> + 0x280700 (0x55e3306d4700 in /usr/bin/python3.10)
frame #41: _PyObject_MakeTpCall + 0x25b (0x55e3305a4a7b in /usr/bin/python3.10)
frame #42: _PyEval_EvalFrameDefault + 0x64e6 (0x55e33059d096 in /usr/bin/python3.10)
frame #43: _PyFunction_Vectorcall + 0x7c (0x55e3305ae9fc in /usr/bin/python3.10)
frame #44: PyObject_Call + 0x122 (0x55e3305bd492 in /usr/bin/python3.10)
frame #45: _PyEval_EvalFrameDefault + 0x2a27 (0x55e3305995d7 in /usr/bin/python3.10)
frame #46: _PyFunction_Vectorcall + 0x7c (0x55e3305ae9fc in /usr/bin/python3.10)
frame #47: _PyEval_EvalFrameDefault + 0x2a27 (0x55e3305995d7 in /usr/bin/python3.10)
frame #48: _PyFunction_Vectorcall + 0x7c (0x55e3305ae9fc in /usr/bin/python3.10)
frame #49: _PyEval_EvalFrameDefault + 0x614a (0x55e33059ccfa in /usr/bin/python3.10)
frame #50: <unknown function> + 0x1687f1 (0x55e3305bc7f1 in /usr/bin/python3.10)
frame #51: _PyEval_EvalFrameDefault + 0x614a (0x55e33059ccfa in /usr/bin/python3.10)
frame #52: _PyFunction_Vectorcall + 0x7c (0x55e3305ae9fc in /usr/bin/python3.10)
frame #53: _PyObject_FastCallDictTstate + 0x16d (0x55e3305a3cbd in /usr/bin/python3.10)
frame #54: _PyObject_Call_Prepend + 0x5c (0x55e3305b986c in /usr/bin/python3.10)
frame #55: <unknown function> + 0x280700 (0x55e3306d4700 in /usr/bin/python3.10)
frame #56: _PyObject_MakeTpCall + 0x25b (0x55e3305a4a7b in /usr/bin/python3.10)
frame #57: _PyEval_EvalFrameDefault + 0x75a0 (0x55e33059e150 in /usr/bin/python3.10)
frame #58: _PyFunction_Vectorcall + 0x7c (0x55e3305ae9fc in /usr/bin/python3.10)
frame #59: _PyEval_EvalFrameDefault + 0x8ac (0x55e33059745c in /usr/bin/python3.10)
frame #60: _PyFunction_Vectorcall + 0x7c (0x55e3305ae9fc in /usr/bin/python3.10)
frame #61: _PyEval_EvalFrameDefault + 0x2a27 (0x55e3305995d7 in /usr/bin/python3.10)
frame #62: _PyFunction_Vectorcall + 0x7c (0x55e3305ae9fc in /usr/bin/python3.10)
frame #63: _PyEval_EvalFrameDefault + 0x614a (0x55e33059ccfa in /usr/bin/python3.10)
```

### PR DESCRIPTION
**Pull request by wujingyue**
Fixes #2552. 

CI: https://gitlab-master.nvidia.com/dl/pytorch/fuser-gh-mirror/-/pipelines/16451691

### CODE REVIEW

**Comment by wujingyue** [comment]
!build

File: csrc/ops/alias.cpp [hunk 1]
```diff
@@ -790,7 +790,8 @@ TensorView* slice(TensorView* inp, const std::vector<Slice>& ranges) {
       out_rf_id = IterDomain::resize(
           out_root_id,
           SimplifyingIrBuilder::negExpr(range.start),
-          SimplifyingIrBuilder::subExpr(range.stop, inp_root_id->extent()),
+          SimplifyingIrBuilder::subExpr(
+              range.stop, inp_root_id->getMaybeExpandedExtent()),
```
Comment (jjsjann123) [inline csrc/ops/alias.cpp:794]: I think you also need to update line #777 for the normalize_slice_range.

File: tests/cpp/test_alias.cpp [hunk 2]
```diff
@@ -1395,4 +1395,28 @@ TEST_F(AliasTest, SegmentMetaOps) {
...
+TEST_F(AliasTest, Issue2552) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  TensorView* x = makeContigConcreteTensor({1, 3});
+  TensorView* y = makeContigConcreteTensor({1, 3});
+  fusion->addInput(x);
+  fusion->addInput(y);
+  x = expand(x, {IrBuilder::create<Val>(2), x->axis(1)->extent()});
+  x = slice(x, /*starts=*/{0, 0}, /*stops=*/{1, 3});
```
Comment (jjsjann123) [inline tests/cpp/test_alias.cpp:1407]: This works for the example. I tried to play with a variation where we expand to something big and slice down to a non-broadcast size, this seems to break for me.

```
  x = expand(x, {IrBuilder::create<Val>(8), x->axis(1)->extent()});
  x = slice(x, /*starts=*/{0, 0}, /*stops=*/{5, 3});
```
Can we also try something like this?

**Review by jjsjann123 — APPROVED** [review]
stamping as-is since this fixes an existing issue.

Some comment for a follow up PR maybe?

**Comment by wujingyue** [comment]
@xwang233, FYI, smoke_test_wheel_stable_17 seems to break regardless of this PR: https://gitlab-master.nvidia.com/dl/pytorch/fuser-gh-mirror/-/jobs/100639742 is

File: tests/cpp/test_alias.cpp (hunk 2 above)
Comment (wujingyue) [reply tests/cpp/test_alias.cpp:1407]: Indeed. Sounds like a different bug, in MarkAliasesPrepare: 

```
C++ exception with description "The contiguity of a broadcast/reduction dimension must be None. The contiguity of a non-broadcast/reduction dimension must be true/false
Exception raised from validateContiguity at /opt/pytorch/nvfuser/csrc/ir/nodes.cpp:2960 (most recent call first):
```

Comment (jjsjann123) [reply tests/cpp/test_alias.cpp:1407]: Feel free to merge this PR as-is and we (or I) can follow up in a different PR.

**Comment by wujingyue** [comment]
!build

File: tests/cpp/test_alias.cpp (hunk 2 above)
Comment (wujingyue) [reply tests/cpp/test_alias.cpp]: #2555

**Pull request merged**